CHANGELOG.md

```diff
@@ -3,6 +3,26 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
 
+## [2.3.0] - 2023-11-23
+### :magic_wand: Added
+- Endpoint override for secrets manager plugin ([PR #707](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/707)).
+- Allow users to set up a lambda to initialize new connections ([PR #705](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/705)).
+- Introduced Dialect.prepareConnectProperties ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).
+- Native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
```

Comment:
Didn't add this in line, but should we consider "Native Open Telemetry support"?

```suggestion
- Native telemetry support ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
```

Reply:
Telemetry is the feature we are adding, and OpenTelemetry is a standard that we are using to enable Telemetry. The documentation specifies it - we are adding native support for instrumentation based on the OpenTelemetry standard - not sure if we should mention OpenTelemetry here

Reply:
That's fine, let's keep it with your original proposal.